core/node/node_framework/src/service/mod.rs

```diff
@@ -59,10 +59,19 @@ impl ZkStackServiceBuilder {
             .enable_all()
             .build()
             .unwrap();
-        Ok(Self {
+        Ok(Self::on_runtime(runtime))
+    }
+
+    /// Creates a new builder with the provided Tokio runtime.
+    /// This method can be used if asynchronous tasks must be performed before the service is built.
+    ///
+    /// However, it is not recommended to use this method to spawn any tasks that will not be managed
+    /// by the service itself, so whenever it can be avoided, using [`ZkStackServiceBuilder::new`] is preferred.
+    pub fn on_runtime(runtime: Runtime) -> Self {
```

Comment:
Dumb question: Will this method be used in the future? AFAIU, it's unused now, other than the call above. Should it be used inside `ExternalNodeBuilder::new()`?

Reply:
Oh, that's a great catch. Yes, it should be called there.